src/operator-docs/transformation/concatMap.ts

```diff
@@ -1,6 +1,78 @@
 import { OperatorDoc } from '../operator.model';
 
 export const concatMap: OperatorDoc = {
-  'name': 'concatMap',
-  'operatorType': 'transformation'
+  name: 'concatMap',
+  operatorType: 'transformation',
+  signature: `concatMap(project: (value: T, index: number) =>  ObservableInput<I>,
+  ?resultSelector: (outerValue: T, innerValue: I, outerIndex: number, innerIndex: number) => R): Observable`,
+  parameters: [
+    {
+      name: 'project',
+      type: 'function(value: T, index: number): ObservableInput',
+      attribute: '',
+      description: `A function that, when applied to an item emitted by the source
+       Observable, returns an Observable.`
+    },
+    {
+      name: 'resultSelector',
+      type:
+        'function(outerValue: T, innerValue: I, outerIndex: number, innerIndex: number): any',
+      attribute: 'optional',
+      description: `A function to produce the value on the output Observable based on the values
+      and the indices of the source (outer) emission and the inner Observable
+      emission. The arguments passed to this function are:
+      - 'outerValue': the value that came from the source.
+      - 'innerValue': the value that came from the projected Observable.
+      - 'outerIndex': the "index" of the value that came from the source.
+      - 'innerIndex': the "index" of the value from the projected Observable.`
+    }
+  ],
+  marbleUrl: 'http://reactivex.io/rxjs/img/concatMap.png',
+  shortDescription: {
+    description: `Project each source value to an Observable which is merge in the output observable,
+    in a serialized fashion waiting for each one to complete before merging the next one
+    <span class="informal">Maps each value to an Observable, then flattens all of
+    these inner Observables using <a href="#/operators/concatAll">concatAll</a>.</span>`
+  },
+  walkthrough: {
+    description: `the source observable maps values to inner observable, subscribe and emit in order.
+     After subscribing the source observable is ended therefore the concatMap's project function
+     is only executed once. the second parameter 'resultFunction', allows you to access to the index of
+     source observable and inner observable (besides the items)`
+  },
+  examples: [
```

Comment:
I'm wondering if a better starting example for `concatMap` would be showing delayed observables executing in order, after completion. Generally, questions I see about `concatMap` are about why you would want to use this vs `mergeMap` or `switchMap`. I think we need to emphasize the difference between these three in the primary examples. What do you all think? @sumitarora @ashwin-sureshkumar @ladyleet

Reply:
But I think this example could works to prove the singularity of concatMap:
with concatMap the project Function is executed just one time:
Click(0), Interval(0) -> Click(0), Interval(1) -> Click(0), Interval(2) etc

with mergeMap the project Function is executed every time I make a click (it could produce memory leaks):
Click(0), Interval(0) -> (click) Click(0), Interval(1) - Click(1), Interval(0) -> etc

with switchMap the project Function is executed every time I make a click but it ends the active subscription until switch to another observable:
Click(0), Interval(1) -> (click) Click(1), Interval(0) -> (click) Click(2), Interval(0)

Please check the live example, https://stackblitz.com/edit/concatmap?file=index.ts

Reply:
I'm 🤔about this one. I think this example shows how `concatMap` works, but I'm not sure it would clearly show it's use versus `mergeMap` or `switchMap`, to a novice, like @btroncone mentioned.

The first time `concatMap`'s use became apparent to me was when I saw ajax results being returned in order of pagination. I tried switching to a `mergeMap` and saw results returning out of order, because some paginated results were able to return faster than others. That's when `concatMap` clicked for me, as well as concurrency with `mergeMap`.

The example also uses a resultSelector function, and it looks like those are probably going to get the axe: https://github.com/ReactiveX/rxjs/pull/3304 It might be best to omit them from examples on the docs here.